src/program.rs

```diff
@@ -207,6 +217,132 @@ where
     };
 }
 
+/// Create a markdown file with documentation for the specified FileId.
+fn output_doc(cache: &mut Cache, file_id: FileId) -> Result<(), Error> {
+    // Parse the file into the cache
+    cache.parse(file_id)?;
+
+    // For every term (although there should only be one)
+    for (file_id, term) in cache.terms() {
+        // Create a new Markdown Document, we will add our documentation to this document
+        let document = AstNode::from(NodeValue::Document);
+
+        // Our nodes in the Markdown document are owned by this arena
+        let arena = Arena::new();
+
+        // The default ComrakOptions disables all extensions (essentially reducing to CommonMark)
+        let options = ComrakOptions::default();
+
+        // Populate the document with the documentation
+        to_markdown(&term.term, 0, &arena, &document, &options)?;
+
+        // Get the original path of the file we parsed
+        let file_path: &Path = cache.files().name(*file_id).as_ref();
+
+        // We are only interested in the original name
+        let file_name = file_path
+            .file_name()
+            .expect(format!("Could not get file name of: {}", file_path.display()).as_ref());
+
+        // Attempt to create the documentation directory
+        let docpath: &Path = Path::new(".nickel/doc/");
+        fs::create_dir_all(docpath).expect("Could not create documentation destination directory");
```

Comment:
We should restrict `expect` and `unwrap` to things we are sure really happen (or if it does happen, a fundamental invariant is being broken and it's justified to panic). Here this case may very well happen for various reasons and it's not nice to the user to print an austere panic message. There is an `Error::IOError` variant than you can use for that.